hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -253,17 +253,27 @@ private Object proxyOrLoad(
 			// associated with the PC, return it
 			final Object managed = persistenceContext.getEntity( keyToLoad );
 			if ( managed != null ) {
-				return managed;
+				if ( options.isCheckDeleted() ) {
+					final EntityEntry entry = persistenceContext.getEntry( managed );
+					final Status status = entry.getStatus();
+					if ( status == Status.DELETED || status == Status.GONE ) {
+						return null;
+					}
+
+					return managed;
+				}
 			}
 
 			// we won't create proxies, but if there is already one associated with
 			// the PC we should use it
-			final Object proxy = persistenceContext.getProxy( keyToLoad );
-			if ( proxy != null ) {
-				return proxy;
+			if ( persister.hasProxy() ) {
```

Comment:
inside this block  `persister.getEntityMetamodel().getBytecodeEnhancementMetadata().isEnhancedForLazyLoading()` have to be `true` so `persister.hasProxy()` so if i do no miss something based on the implementation 
`public boolean hasProxy() {
		// skip proxy instantiation if entity is bytecode enhanced
		return entityMetamodel.isLazy() && !entityMetamodel.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading();
	}`
it will be always `false`

Reply:
* `#hasProxy` - can it produce HibernateProxy instances
* `#isEnhancedForLazyLoading` - is the entity enhanced